platooning_strategic_IHP/src/platoon_manager_ihp.cpp

```diff
@@ -631,4 +631,63 @@ namespace platoon_strategic_ihp
         // ---> 4.4 return IHP calculated desired speed
         return pos_des;
     }
+
+    // UCLA: find the clsest index to the cut-in joining vehicle.
+    // Note: The index pointing to the gap-leading vehicle's index.
+    int PlatoonManager::getClosestIndex(double joinerDtD)
+    {   
+        /*
+            A naive way to find the closest member to the jioning vehicle.
+            Note: The potential gap leading vehicle should be in front of the joiner (i.e., gap leading vehicle's dtd > joiner's dtd).
+                  If the joiner is already in front of leader --> all loop result will be negative --> return -1 (i.e., cut-in front) 
+        */
+
+        double min_diff = 99999.000;
+        int cut_in_index = -1;
+        // Loop through all platoon members  
+        for(int i = 0; i < platoon.size(); i++) 
+        {
+            double current_member_dtd = platoon[i].vehiclePosition; 
+            double curent_dtd_diff = current_member_dtd - joinerDtD;
+            // update min index
+            if (curent_dtd_diff > 0 && curent_dtd_diff < min_diff)
+                {
+                    min_diff = current_member_dtd;
+                    cut_in_index = i;
+                }
+        }
+        
+        return cut_in_index;
+    }
+
+    // UCLA: find the cut-in join target gap size in downtrack distance (m).
```

Comment:
General approach to this method sometimes considers host vehicle length and sometimes not.  What assumption is being made about the origin of the vehicle coordinate frame?  That is, to what point on the vehicle is DTD measured?  This assumption should be stated in a comment.  Elsewhere in Carma, the vehicle's origin is at the rear axle, and I imagine that applies here also.

Use of vehicle length is appropriate in the middle cut-in case.  However, it seems it should apply to at least one of the other two cases as well.  For example, if the origin is at front bumper, then the front cut-in case should subtract vehicle length from the DTD diff.  Unfortunately, having an origin somewhere in the middle of the vehicle makes it difficult to do this math.  But Maybe it could be ignored for the front join case (origin near the rear), but it should be subtracted in the rear join case.